crates/re_ui/src/ui_ext.rs

```diff
@@ -1153,3 +1054,118 @@ impl UiExt for egui::Ui {
+                        let icon_width = ui.spacing().icon_width_inner;
+                        let space_after_icon = ui.spacing().icon_spacing;
+
+                        let mut layout_job = egui::WidgetText::from(label).into_layout_job(
+                            ui.style(),
+                            egui::FontSelection::Default,
+                            egui::Align::LEFT,
+                        );
+                        layout_job.wrap = TextWrapping::truncate_at_width(
+                            header_size_without_button.x
+                                - (space_before_icon + icon_width + space_after_icon),
+                        );
+                        let galley = ui.fonts(|fonts| fonts.layout_job(layout_job));
+
+                        let header_response =
+                            ui.allocate_response(header_size_without_button, egui::Sense::click());
+                        let rect = header_response.rect;
+
+                        let icon_rect = egui::Rect::from_center_size(
+                            header_response.rect.left_center()
+                                + egui::vec2(space_before_icon + icon_width / 2.0, 0.0),
+                            egui::Vec2::splat(icon_width),
+                        );
+                        let icon_response = header_response.clone().with_new_rect(icon_rect);
+                        ui.paint_collapsing_triangle(
+                            openness,
+                            icon_rect.center(),
+                            ui.style().interact(&icon_response),
+                        );
+
+                        let visuals = ui.style().interact(&header_response);
+
+                        let optical_vertical_alignment = 0.5; // improves perceived vertical alignment
+                        let text_pos = icon_response.rect.right_center()
+                            + egui::vec2(
+                                space_after_icon,
+                                -0.5 * galley.size().y + optical_vertical_alignment,
+                            );
+
+                        ui.painter().galley(text_pos, galley, visuals.text_color());
+
+                        // Let the rect cover the full panel width:
+                        let bg_rect = egui::Rect::from_x_y_ranges(ui.full_span(), rect.y_range());
+
+                        ui.painter().set(
+                            background_frame,
+                            Shape::rect_filled(bg_rect, 0.0, visuals.bg_fill),
+                        );
+
+                        if header_response.clicked() {
+                            state.toggle(ui);
+                        }
+                    },
+                );
+            },
+        );
+    });
+
+    state.show_body_unindented(ui, |ui| {
+        ui.add_space(4.0); // Add space only if there is a body to make minimized headers stick together.
+        add_body(ui);
+        ui.add_space(4.0); // Same here
+    });
+}
```

Comment:
it would been a lot easier to review if the moving of the code and the changing of it were two different commits

Reply:
ok, then I can review it all I guess… except there has been some rebase weirdness now

Reply:
I just rebased it on main. It was originally chained to #6605, which has just been merged.